frontend/src/lib/components/IntervalFilter/intervals.ts

```diff
@@ -1,20 +1,32 @@
-export const intervals = {
+export type IntervalKeyType = 'hour' | 'day' | 'week' | 'month'
+
+export type Intervals = {
+    [key in IntervalKeyType]: {
+        label: string
+        newDateFrom?: string
+        disabledReason: string | undefined
+    }
+}
+
+export const intervals: Intervals = {
     hour: {
         label: 'Hour',
         newDateFrom: 'dStart',
+        disabledReason: undefined,
```

Comment:
We can omit these

Reply:
I wanted to be explicit here so that it's clear that this can be here although that's what the type is for :D 

This didn't use to have a separate type, that's why